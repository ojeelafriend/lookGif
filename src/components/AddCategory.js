import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputValue = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((category) => {
				return [inputValue, ...category];
			});
			setInputValue('');
		}
	};

	return (
		<>
			<form onSubmit={handleSubmit}>
				<input
					type="text"
					value={inputValue}
					placeholder="press enter once you type"
					onChange={handleInputValue}
				/>
			</form>
			<p hidden={true}>{inputValue}</p>
		</>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
