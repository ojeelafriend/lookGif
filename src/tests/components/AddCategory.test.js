import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import AddCategory from '../../components/AddCategory';

describe('Test <AddCategory/>', () => {
	const setCategories = jest.fn();

	let wrapper;

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

	test('Success render <AddCategory/>', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('OnChange receive success', () => {
		const input = wrapper.find('input');
		const value = 'Test success';

		input.simulate('change', { target: { value } });

		expect(wrapper.find('p').text().trim()).toBe(value);
	});
	test('Not post onSubmit in the setCategories', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategories).not.toHaveBeenCalled();
	});
	test('If the input has word then post setCategories', () => {
		wrapper.find('input').simulate('change', { target: { value: 'One piece' } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategories).toHaveBeenCalled();
		expect(wrapper.find('p').text()).toBe('');
	});
	test('Call setCategories how function', () => {
		wrapper.find('input').simulate('change', { target: { value: 'One piece' } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });

		expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
	});
});
