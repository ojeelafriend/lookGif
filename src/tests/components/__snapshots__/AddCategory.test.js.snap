// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory/> Success render <AddCategory/> 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="press enter once you type"
      type="text"
      value=""
    />
  </form>
  <p
    hidden={true}
  />
</Fragment>
`;
